### PR TITLE
feat: ISO VolumeID named with ReleaseID

### PR DESCRIPTION
**Pull request by xerc**
Fixes https://github.com/siderolabs/talos/issues/8769

### CODE REVIEW

**Comment by smira** [comment]
Thanks for putting this together, I will do some changes to make this work only for Talos >= 1.8.0 (to keep old ISOs reproducible).

**Comment by smira** [comment]
```
$ file _out/metal-amd64.iso
_out/metal-amd64.iso: ISO 9660 CD-ROM filesystem data (DOS/MBR boot sector) 'TALOS_V1_8_0_ALPHA_1_66_G81DA98A' (bootable)
```

```
$ isoinfo -i _out/metal-amd64.iso -d
CD-ROM is in ISO 9660 format
System id: 
Volume id: TALOS_V1_8_0_ALPHA_1_66_G81DA98A
Volume set id: Talos-v1.8.0-alpha.1-66-g81da98ac45-dirty
```

```
$ file _out/metal-amd64-secureboot.iso
_out/metal-amd64-secureboot.iso: ISO 9660 CD-ROM filesystem data 'TALOS_SB_V1_8_0_ALPHA_1_66_G81DA' (bootable)
```

```
$ isoinfo -i _out/metal-amd64-secureboot.iso -d 
CD-ROM is in ISO 9660 format
System id: 
Volume id: TALOS_SB_V1_8_0_ALPHA_1_66_G81DA
Volume set id: Talos-SB-v1.8.0-alpha.1-66-g81da98ac45-dirty
```

File: Makefile [hunk 1]
```diff
@@ -378,7 +378,7 @@ image-%: ## Builds the specified image. Valid options are aws, azure, digital-oc
 	@docker pull $(REGISTRY_AND_USERNAME)/imager:$(IMAGE_TAG)
 	@for platform in $(subst $(,),$(space),$(PLATFORM)); do \
 		arch=$$(basename "$${platform}") && \
-		docker run --rm -t -v /dev:/dev -v $(PWD)/$(ARTIFACTS):/secureboot:ro -v $(PWD)/$(ARTIFACTS):/out --network=host --privileged $(REGISTRY_AND_USERNAME)/imager:$(IMAGE_TAG) $* --arch $$arch $(IMAGER_ARGS) ; \
+		docker run --rm -t -v /dev:/dev -v $(PWD)/$(ARTIFACTS):/secureboot:ro -v $(PWD)/$(ARTIFACTS):/out -e SOURCE_DATE_EPOCH=$(SOURCE_DATE_EPOCH) --network=host --privileged $(REGISTRY_AND_USERNAME)/imager:$(IMAGE_TAG) $* --arch $$arch $(IMAGER_ARGS) ; \
```
Comment (smira) [inline Makefile:381]: this was a bug that we don't pass `SOURCE_DATA_EPOCH`

**Review by frezbo — APPROVED** [review]

**Pull request merged**